File: nowa/2019/c++/zad4_1.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <vector>
using namespace std;

// funkcja sprawdzajaca czy dana liczba jest potega liczby 3
bool isPowerOfThree(int number)
{
    while (number % 3 == 0)
        number /= 3;

    if (number == 1)
        return true;
    else
        return false;
}

string zad4_1()
{
    string line;
    vector<int> numbers;

    // wczytywanie danych z pliku
    fstream file("../dane/liczby.txt");

    if (file.is_open())
    {
        while (getline(file, line))
            numbers.push_back(atoi(line.c_str()));
    }
    file.close();

    int answer = 0;

    // przechodzenie po wszystkich danych nam liczbach i sprawedzanie czy sa potegami 3
    for (int i = 0; i < numbers.size(); i++)
    {
        if (isPowerOfThree(numbers[i]))
            answer++;
    }

    return "4.1. Liczba elementow, ktore sa potegami 3: " + to_string(answer);
}

File: nowa/2019/python/zad4_1.py
# zaladuj dane
data = [int(x[:-1]) for x in open('../dane/liczby.txt')]


def is_power_of(x, n):
    """sprawdza czy x jest potega liczby n"""

    # najpierw wyeliminowac 2 przypadki gdzie wiadomo ze liczba napewno spelnia lub nie spelnia warukow
    if x == 1:
        return True
    if x % n != 0:
        return False
    # dzielic liczbe x az bedzie sie rownala (badz nie) n
    while x >= n:
        if x == n:
            return True
        x /= n
    return False


count = 0

# przejdz po danych i policz potegi liczby 3
for x in data:
    count += is_power_of(x, 3)

# zapisz odpowiedz
answer = f'4.1. Ilosc liczb ktore sa potegami liczby 3: {count}'
print(answer)
